tests/integration/worker/test_submission_worker.py

```diff
@@ -0,0 +1,394 @@
+            created_by=self.user)
+
+        self.participant = Participant.objects.create(
+            user=self.user,
+            status=Participant.SELF,
+            team=self.participant_team)
+
+        self.challenge = Challenge.objects.create(
+            title='Test Challenge',
+            description='Description for test challenge',
+            terms_and_conditions='Terms and conditions for test challenge',
+            submission_guidelines='Submission guidelines for test challenge',
+            creator=self.challenge_host_team,
+            start_date=timezone.now(),
+            end_date=timezone.now() + timedelta(days=1),
+            published=True,
+            approved_by_admin=True,
+            enable_forum=True,
+            anonymous_leaderboard=False)
+
+        try:
+            os.makedirs("/tmp/evalai")
+        except OSError:
+            pass
+
+        with self.settings(MEDIA_ROOT='/tmp/evalai'):
+            self.challenge_phase = ChallengePhase.objects.create(
+                name='Challenge Phase',
+                description='Description for Challenge Phase',
+                leaderboard_public=False,
+                is_public=True,
+                start_date=timezone.now(),
+                end_date=timezone.now() + timedelta(days=1),
+                challenge=self.challenge,
+                test_annotation=SimpleUploadedFile('test_sample_file.txt',
+                                                   b'Dummy file content',
+                                                   content_type='text/plain')
+            )
+        self.url = reverse_lazy('jobs:challenge_submission',
+                                kwargs={'challenge_id': self.challenge.pk,
+                                        'challenge_phase_id': self.challenge_phase.pk})
+
+        self.submission = Submission.objects.create(
+            participant_team=self.participant_team,
+            challenge_phase=self.challenge_phase,
+            created_by=self.challenge_host_team.created_by,
+            status="submitted",
+            input_file=SimpleUploadedFile(
+                'user_annotation_file.txt',
+                b"Dummy file content",
+                content_type='text/plain'
+            ),
+            method_name="Test Method",
+            method_description="Test Description",
+            project_url="http://testserver/",
+            publication_url="http://testserver/",
+            is_public=True,
+        )
+
+    def tearDown(self):
+        try:
+            shutil.rmtree('/tmp/evalai')
+        except OSError:
+            pass
+
+
+class ProcessSubmissionCallbackTestClass(BaseTestClass):
+    @mock.patch("scripts.workers.submission_worker.process_submission_message")
+    def test_process_submission_callback(self, mock_psm):
+        message = {
+            "challenge_pk": self.challenge.pk,
+            "phase_pk": self.challenge_phase.pk,
+            "submission_pk": self.submission.id
```

Comment:
Please follow one convention either `pk` or `id`. I think we were following `pk`.